server/middleware/passport.js

```diff
@@ -28,173 +29,158 @@ passport.deserializeUser((id, done) => {
 });
 
 passport.use('local-signup', new LocalStrategy({
-  usernameField: 'email',
+  usernameField: 'username',
   passwordField: 'password',
   passReqToCallback: true
 },
-  (req, email, password, done) => {
-    // check to see if there is any account with this email address
-    return models.Profile.where({ email }).fetch()
-      .then(profile => {
-        // create a new profile if a profile does not exist
-        if (!profile) {
-          return models.Profile.forge({ email }).save();
+  (req, username, password, done) => {
+    return User.getByUsername(username)
+      .then((user) => {
+        if (user) {
+          return done(null, false, {
+            message: 'Username is already taken'
+          });
+        } else {
+          let generateHash = (password) => {
+            return bCrypt.hashSync(password, bCrypt.genSaltSync(8), null);
+          };
+          let data = {
+            username,
+            password: generateHash(password),
+            firstname: req.body.firstname,
+            lastname: req.body.lastname
+          };
+          User.create(data)
+            .then((newUser) => {
+              if (!newUser) {
+                return done(null, false);
+              } else {
+                return done(null, newUser);
+              }
+            })
+            .catch((err) => {
+              return done(null, false, {
+                message: 'Username is invalid'
+              });
+            });
         }
-        // throw if any auth account already exists
-        if (profile) {
-          throw profile;
-        }
-
-        return profile;
-      })
-      .tap(profile => {
-        // create a new local auth account with the user's profile id
-        return models.Auth.forge({
-          password,
-          type: 'local',
-          profile_id: profile.get('id')
-        }).save();
-      })
-      .then(profile => {
-        // serialize profile for session
-        done(null, profile.serialize());
-      })
-      .error(error => {
-        done(error, null);
-      })
-      .catch(() => {
-        done(null, false, req.flash('signupMessage', 'An account with this email address already exists.'));
       });
   }));
 
 passport.use('local-login', new LocalStrategy({
-  usernameField: 'email',
+  usernameField: 'username',
```

Comment:
Changes like this should be in a different PR. Please do not try to fix bugs or refactor in the same PR you are adding features.